go-controller/pkg/node/gateway_init.go

```diff
@@ -114,7 +114,7 @@ func (n *OvnNode) initGateway(subnet *net.IPNet, nodeAnnotator kube.Annotator,
 	}
 
 	// Wait for gateway resources to be created by the master
-	waiter.AddWait(gatewayReady, prFn)
+	waiter.AddWait(gatewayReady, prFn, util.ForceOVNRecompute)
```

Comment:
This seems like a weird way to incorporate this, particularly since you pass the same reconciler in every `waiter.AddWait` call. I'd make the reconciler be defined as part of the `startupWaiter` itself rather than being per-`waitTask`.